R/plot_dot.R

```diff
@@ -885,7 +883,8 @@ extract_sm_nodes <- function(model, theme, structure_only = FALSE) {
 
     if (startsWith(construct_type, "HOC") && !structure_only) {
 
-      row_index <- grepl(construct, model$mmMatrix[,1])
+      # row_index <- grepl(construct, model$mmMatrix[,1])
+      row_index <- model$mmMatrix[,1] == construct
```

Comment:
Can we use a helper function to find contruct in mmMatrix

Reply:
@NicholasDanks we can def refactor this one by using/writing a helper function that gets all the construct names from the measurement model. It might be `mm_constructs` or `all_construct_names` (take a look at both). Or else just make a wrapper function for the expression `model$mmMatrix[,1]`.

Reply:
and feel free to assign this back to me!